lib/opentelemetry/trace/exporter/circuit.lua

```diff
@@ -0,0 +1,77 @@
+local util = require("opentelemetry.util")
+local otel_global =  require("opentelemetry.global")
+
+local _M = {
+    OPEN = "open",
+    CLOSED = "closed",
+    HALF_OPEN = "half-open"
+}
+
+local mt = {
+    __index = _M
+}
+
+
+function _M.new(options)
+    options = options or {}
+    local self = {
+        circuit_reset_timeout_ms = options["circuit_reset_timeout_ms"] or 5000,
+        failure_threshold = options["failure_threshold"] or 5,
+        failure_count = 0,
+        open_start_time_ms = nil,
+        state = "closed",
+    }
+    return setmetatable(self, mt)
+end
+
+function _M.should_make_request(self)
```

Comment:
should_make_request -> attempt_request

Reply:
I do not think we should rename to `attempt_request`, since we are not attempting a request in the method. I am open to other names, though!